tests/big-int/ctor_sample_test.cpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Boost Software License - Version 1.0 - August 17th, 2003
+ * Permission is hereby granted, free of charge, to any person
+ * or organization obtaining a copy of the software and
+ * accompanying documentation covered by this license
+ * (the "Software") to use, reproduce, display, distribute,
+ * execute, and transmit the Software, and to prepare derivative
+ * works of the Software, and to permit third-parties to whom the
+ * Software is furnished to do so, all subject to the following:
+ *
+ * The copyright notices in the Software and this entire statement,
+ * including the above license grant, this restriction and the following
+ * disclaimer, must be included in all copies of the Software, in whole or
+ * in part, and all derivative works of the Software, unless such copies
+ * or derivative works are solely in the form of machine-executable
+ * object code generated by a source language processor.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, TITLE AND
+ * NON-INFRINGEMENT. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR ANYONE
+ * DISTRIBUTING THE SOFTWARE BE LIABLE FOR ANY DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#include <bigint.hpp>
+#include <cassert>
+
+int main()
+{
+    // tests default ctor
+    libbig::largeInt a;
+    assert(a == 0);
```

Comment:
This is wrong.
largeInt are strings and you are asserting it with unsigned int, this will fail

if you write `assert(a == "0")` this will also throw an error because you have not overloaded the `==` operator for largeInt and strings

Reply:
tests are misleading.
build is successful because this test file is not executed at the time of testing

Reply:
I wrote the == operator overload with int and long long int. Therefore, `assert(a == 0);` shouldn't have any problem executing, because it calls `libbig::largeInt::operator==(int)`, which I made sure works fine. I know the test files are not automatically executed, that's why I compiled and executed them myself more than once. The assert calls can indeed execute without problem. They also correctly detect errors, that's why `assert(a == 0);` executes without problem but, given a `libbig::largeInt b(500);`, `assert(b == 400);` aborts the program. I will optimize the == operator and make sure the tests are executing like they should.

Reply:
these tests are supposed for ctor not `==` operator. what i want to say is that
`libbig::largeInt a;` will invoke the default ctor and in default ctor we have `this->number = "0"` which is a string so if we `assert(a == 0)` then it will fail for sure as we are asserting a string against an unsigned int.

also i fixed the ctest you can check now ctor_sample_tests are failing.

for `==` operator you are supposed to write a different test file `equal_to.cpp` in `tests/operators`
hope this makes sense

Reply:
```
Test project /home/runner/work/bigINT/build
    Start 1: sample_test2
1/3 Test #1: sample_test2 .....................   Passed    0.00 sec
    Start 2: sample_test1
2/3 Test #2: sample_test1 .....................   Passed    0.00 sec
    Start 3: ctor_sample_test
3/3 Test #3: ctor_sample_test .................***Failed    0.00 sec

67% tests passed, 1 tests failed out of 3
Errors while running CTest

Total Test time (real) =   0.01 sec

The following tests FAILED:
	  3 - ctor_sample_test (Failed)
Error: Process completed with exit code 8.
```

Reply:
I think the issue is in the cmake files, because I successfully compiled it myself with
```batch
$ g++ -Wall -c ./big-int/src/bigint.cpp
$ g++ -Wall -c ./big-int/src/operators/equal_to.cpp
$ g++ -Wall -c ./tests/ctor_sample_test.cpp
$ g++ -Wall bigint.o equal_to.o ctor_sample_test.o
```
It executes the file just fine, I added couts after each assert call and it gives :
First test passed
Second test passed
Third test passed
Fourth test passed
without aborting the program. The default constructor test passes because, even if the default destructor is called, `this->number` has a value ("0"), and it can and will be compared to a string with `assert(a == 0);` because 0 is an int and `libbig::largeInt::operator==(int)` will be called, comparing "0" (`this->number`) with 0 (int). If I replace the assert statement with `assert(a == 1);`, the program will abort. That said, I understand the test file is about both ctors AND the == operator overload and I will try to make it test only ctors.

Reply:
okay let me check it on my system.
can you please mention your gcc version?

Reply:
okay my bad i had not seen the overloaded `==` definition properly